forms.rst

```diff
@@ -449,47 +424,6 @@ the date as a string in the box)::
 .. image:: /_images/form/simple-form-2.png
     :align: center
 
-Each field type has a number of different options that can be passed to it.
-Many of these are specific to the field type and details can be found in
-the documentation for each type.
-
-.. sidebar:: The ``required`` Option
-
-    The most common option is the ``required`` option, which can be applied to
-    any field. By default, the ``required`` option is set to ``true``, meaning
-    that HTML5-ready browsers will apply client-side validation if the field
-    is left blank. If you don't want this behavior, either
-    :ref:`disable HTML5 validation <forms-html5-validation-disable>`
-    or set the ``required`` option on your field to ``false``::
-
-        ->add('dueDate', DateType::class, [
-            'widget' => 'single_text',
-            'required' => false
-        ])
-
-    Also note that setting the ``required`` option to ``true`` will **not**
-    result in server-side validation to be applied. In other words, if a
-    user submits a blank value for the field (either with an old browser
-    or web service, for example), it will be accepted as a valid value unless
-    you use Symfony's ``NotBlank`` or ``NotNull`` validation constraint.
-
-    In other words, the ``required`` option is "nice", but true server-side
-    validation should *always* be used.
-
-.. sidebar:: The ``label`` Option
-
-    The label for the form field can be set using the ``label`` option,
-    which can be applied to any field::
-
-        ->add('dueDate', DateType::class, [
-            'widget' => 'single_text',
-            'label'  => 'Due Date',
-        ])
-
-    The label for a field can also be set in the template rendering the
-    form, see below. If you don't need a label associated to your input,
-    you can disable it by setting its value to ``false``.
-
 .. index::
    single: Forms; Field type guessing
```

Comment:
I would move the whole section about form type (and type options) guessing into a dedicated article. Handling this here in the main article about forms can be confusing.

Reply:
Well, I don't say that this is the perfect solution for everything. I'm just saying that it's better than it was ;-)